tests/data_context/test_get_data_context.py

```diff
@@ -0,0 +1,230 @@
+    )
+
+
+@pytest.mark.unit
+def test_config_returns_base_context_overrides_yml(tmp_path: pathlib.Path):
+    project_path = tmp_path / "empty_data_context"
+    project_path.mkdir()
+    project_path_str = str(project_path)
+    gx.data_context.DataContext.create(project_path_str)
+    context_path = project_path / "great_expectations"
+    context = gx.get_context(context_root_dir=context_path)
+    assert isinstance(context, DataContext)
+    assert context.expectations_store_name == "expectations_store"
+
+    config: DataContextConfig = DataContextConfig(
+        config_version=3.0,
+        plugins_directory=None,
+        evaluation_parameter_store_name="new_evaluation_parameter_store",
+        expectations_store_name="new_expectations_store",
+        datasources={},
+        stores={
+            "new_expectations_store": {"class_name": "ExpectationsStore"},
+            "new_evaluation_parameter_store": {
+                "class_name": "EvaluationParameterStore"
+            },
+            "new_validation_result_store": {"class_name": "ValidationsStore"},
+        },
+        validations_store_name="new_validation_result_store",
+        data_docs_sites={},
+        validation_operators={},
+    )
+    context = gx.get_context(project_config=config, context_root_dir=context_path)
+    assert isinstance(context, BaseDataContext)
+    assert context.expectations_store_name == "new_expectations_store"
+
+
+@pytest.mark.cloud
+@mock.patch("requests.get")
+@mock.patch.dict(
+    os.environ,
+    {
+        "GE_CLOUD_BASE_URL": "http://hello.com",
+        "GE_CLOUD_ORGANIZATION_ID": "1",
+        "GE_CLOUD_ACCESS_TOKEN": "i_am_a_token",
+    },
+    clear=True,
+)
+def test_cloud_config_in_env_returns_cloud_context(
+    mock_request,
+    request_headers,
+):
+    mock_request.return_value.status_code = 200
+    try:
+        assert isinstance(
+            gx.get_context(),
+            CloudDataContext,
+        )
+    except:  # Not concerned with constructor output (only evaluating interaction with requests during __init__)
+        pass
+
+
+@pytest.mark.cloud
+@mock.patch("requests.get")
+@mock.patch.dict(
+    os.environ,
+    {
+        "GE_CLOUD_BASE_URL": "http://hello.com",
+    },
+    clear=True,
+)
+def test_most_cloud_config_in_param_base_url_in_env_returns_cloud_context(
+    mock_request,
+    request_headers,
+):
+    mock_request.return_value.status_code = 200
+    try:
+        assert isinstance(
+            gx.get_context(
+                ge_cloud_organization_id="1", ge_cloud_access_token="i_am_a_token"
+            ),
+            CloudDataContext,
+        )
+    except:  # Not concerned with constructor output (only evaluating interaction with requests during __init__)
+        pass
+
+
+@pytest.mark.cloud
+@mock.patch("requests.get")
+def test_all_cloud_config_in_param_returns_cloud_context(
+    mock_request,
+    request_headers,
+):
+    mock_request.return_value.status_code = 200
```

Comment:
You could refactor this request mock as a fixture and re-use it for all these tests.

```python
@pytest.fixture
def mock_api_200():
   with mock.patch("requests.get") as patch:
         patch.return_value.status_code == 200
         yield patch
```

Reply:
similar point to above. The PR has been updated to `mock.patch` the `retrieve_data_context_config_from_ge_cloud()` method instead, which negates the need for the `try:except`, which I believe was masking an error.